### DISCUSSION TITLE
Autocompletion for comparison on Literals

### DISCUSSION THREAD

**Discussion by bellini666** [microsoft/pylance-release#2742] [Enhancement]
It would be awesome if literals could be used on autocompletions for equals/in/match/etc, like in the following examples:

```python
from typing import Literal

var_type = Literal["foo", "bar", "bin"]


def some_func() -> var_type:
    ...


var = some_func()

if var == "<autocomplete for 'foo', 'bar', 'bin'>":
    ....

if var in ["<autocomplete for 'foo', 'bar', 'bin'>", ...]:
    ....

match (var):
    case "<autocomplete for 'foo', 'bar', 'bin'>":
        ...
```

**Comment by erictraut** [comment]
Since this isn't a core type checking feature request, I'm going to transfer this to the pylance issue tracker.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by multimeric** [comment]
I think this is mostly a duplicate of #2635, as well.
